lib/vm.js

```diff
@@ -85,14 +85,30 @@ function validateString(prop, propName) {
                                'string', prop);
 }
 
+function validateBool(prop, propName) {
+  if (prop !== undefined && typeof prop !== 'boolean')
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               'boolean', prop);
+}
+
 function getContextOptions(options) {
   if (options) {
     const contextOptions = {
       name: options.contextName,
-      origin: options.contextOrigin
+      origin: options.contextOrigin,
+      codeGeneration: options.contextCodeGeneration ? {
```

Comment:
The `typeof options.contextCodeGeneration === 'object'` check should be kept though.